### DISCUSSION TITLE
PC version

### DISCUSSION THREAD

**Discussion by AliROCK66** [GFW-knocker/MahsaNG#25] [Ideas]
درود
امکان داره در آینده نسخه پی سی ساخته بشه ؟حیف این برنامه هست فقط برای اندروید باشه

**Comment by RezaMLS** [comment]
واقعا اگر ساخته بشه عالی میشه میترکونه

**Comment by arshiaojf** [comment]
واقعا به ورژن ویندوز مهسا نیاز هست بنظرم

**Comment by roozbeh9** [comment]
برای لینوکس و ویندوز لطقا پکیج بسازید

**Comment by kamran-kianloo** [comment]
درود بر شما
بی نهایت بهش نیاز داریم

**Comment by maharablab-cmd** [comment]
با تشکر از زحمات این تیم.
لطفا نسخه ویندوز را هم بسازید چون خیلی نیاز هست و هیچ ابزاری به پای این نرم افزار خوب نمیرسه

**Comment by KotikMur999** [comment]
I agree, we need a Windows version. Thanks 🙏

**Comment by MatinEbrahimkhani** [comment]
Dear Team,
I would like to contribute to your project for creating MahsaNG for both Windows and Linux.
your contribution to connecting Iranians to the rest of the world is astonishing, and would like to help you in this matter.

**Comment by slabs37** [comment]
For now i am using MahsaNG on Windows by running it in proxy mode on my phone, then using [tun2proxy](https://github.com/tun2proxy/tun2proxy) to tunnel everything through it

**Reply by MatinEbrahimkhani** [reply]
if it works it works. but a native support would be way better. here is another solution that i use in Manjaro Linux that i don't know if it works for windows.
Hiddify has a Mahsa subscription that loads the mahsang connections to my experience

**Comment by 1r7in0** [comment]
MahsaNG is currently not available for PC platforms.

But MahsaNG uses MahsaNet subscriptions underneath. The public SSOT for those free subscriptions/configurations is **https://github.com/mahsanet/MahsaFreeConfig**.

These subscriptions are the actual core configurations (the "VPN" part). You can use them directly in any V2Ray/Xray client such as v2rayN, Nekoray, Hiddify, Clash Meta, etc.

Main active subscription/configuration directories:

- `mci/` - often works better on MCI networks

- `mtn/` - often works better on MTN networks

- `app/`

- `telegram/` - Telegram-specific (different structure)

The files are refreshed frequently. Each update contains new unique configs (older ones remain available in the git history).

Files in `mci/` and `mtn/` are Base64-encoded lists of share links. They cannot be used directly, you must decode them first:

Clone or download the repository.

Change directory to root of the repository.

Decode the Base64 content:

PowerShell (Windows):
```powershell
# MCI
[System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String((Get-Content .\mci\sub_1.txt -Raw)))

# MTN
[System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String((Get-Content .\mtn\sub_1.txt -Raw)))
```

Bash (Linux/macOS):
```bash
# MCI
base64 -d mci/sub_1.txt

# MTN
base64 -d mtn/sub_1.txt
```

Copy the decoded output and paste it into your client via Configuration > Import Share Links from clipboard.

**Comment by 1r7in0** [comment]
You can use tools like PdaNet+ or EasyTether to fully tunnel your PC connection through your phone's VPN.
Normal USB tethering doesn't include VPN connection to tether.
